### DISCUSSION TITLE
Multiple graphql endpoints

### DISCUSSION THREAD

**Discussion by adamjw3** [apollographql/apollo-server#4500] [General]
Looking at using apollo-server-micro with nextjs in the api folder. 
I will have multiple end points to get data from and do queries, mutations, all the graphql goodness.

e.g end points

https://headlesscmsgraphqlendpoint
https://someotherservicegraphqlendpoint

Any examples, best practices, the easiest way on doing this?

**Comment by abernix** [comment]
Keeping in mind that we highly recommend [having a single graph for an organization](https://principledgraphql.com/integrity#1-one-graph), you can have multiple Apollo Server's running on different paths on a domain/port.

By default, `apollo-server` uses `/` and the integration packages (e.g., `apollo-server-*`) use `/graphql`, but you can specify a different `path`.  

For `apollo-server`, it would be specified as `new ApolloServer({ path: '/different' });` and for `apollo-server-*` it would be done using the `server.applyMiddleware({ path: '/different' })` option.

**Comment by adamjw3** [comment]
@abernix Thanks for your reply.

What if i'm using multiple 3rd party graphql apis (shopify and contentful)

Any examples you can share?

**Comment by adamjw3** [comment]
@abernix with what you suggested i'd have to rewrite the schema for the 3rd party api even though its already graphql.
Think that issue is outlined here.

https://spectrum.chat/apollo/apollo-federation/remote-schemas-and-apollo-federation~74f89ac9-2623-4c51-bb87-8880a140fc27

Any examples of adding multiple 3rd party apis into one using apollo server?
